sourcespec/ssp_build_spectra.py

```diff
@@ -361,7 +389,7 @@ def _build_weight_from_frequency(config, spec):
 
 def _build_weight_from_noise(config, spec, specnoise):
     weight = spec.copy()
-    if specnoise is not None:
+    if specnoise is not None and not np.all(specnoise.data == 0):
```

Comment:
What happens if `np.all(specnoise.data == 0)`?

Reply:
The case `np.all(specnoise.data == 0)` should be equivalent to `specnoise is None`, i.e. noise spectrum is empty. Actually, I'm not sure the latter case ever occurs. The former will occur if the available noise length is zero and the noise window is zero-padded to the length of the signal window. In either case, the else block should be executed.
The code could be made a little more clear by swapping the blocks below if/else, and changing the if condition to:
`if specnoise is None or np.all(specnoise.data == 0):`

Reply:
Agree, pleas make this change.

Reply:
Done